yahp/hparams.py

```diff
@@ -325,6 +329,76 @@ def validate(self):
             category=DeprecationWarning,
         )
 
+    @classmethod
+    def get_json_schema(cls) -> Dict[str, Any]:
+        """Generates and returns a JSONSchema dictionary."""
+
+        res = {
+            'type': 'object',
+            'properties': {},
+            'required': [],
+            'additionalProperties': False,
+        }
+        class_type_hints = get_type_hints(cls)
+        for f in fields(cls):
+            if not f.init:
+                continue
+
+            # Required field
+            if type_helpers.is_field_required(f):
+                res['required'].append(f.name)
+
+            hparams_type = type_helpers.HparamsType(class_type_hints[f.name])
+            # Name is found in registry, set possible values as types in a union type
+            if cls.hparams_registry and f.name in cls.hparams_registry and len(cls.hparams_registry[f.name].keys()) > 0:
+                res['properties'][f.name] = get_registry_json_schema(hparams_type, cls.hparams_registry[f.name])
+            else:
+                res['properties'][f.name] = get_type_json_schema(hparams_type)
+            res['properties'][f.name]['description'] = f.metadata['doc']
+
+        # Delete required list if no fields are required
+        if len(res['required']) == 0:
+            del res['required']
+
+        return res
+
+    @classmethod
+    def dump_jsonschema(cls, f: Union[TextIO, str, pathlib.Path]):
+        """Dump the JSONSchema to ``f``."""
+        if isinstance(f, TextIO) or isinstance(f, TextIOWrapper):
+            json.dump(cls.get_json_schema(), f)
+        else:
+            with open(f, 'w') as file:
+                json.dump(cls.get_json_schema(), file)
+
+    @classmethod
+    def validate_yaml(cls: Type[THparams],
+                      f: Union[str, None, TextIO, pathlib.PurePath] = None,
+                      data: Optional[str] = None):
```

Comment:
To match the `create` API, can data be a dictionary?

```suggestion
                      data: Optional[Dict[str, Any]] = None):
```